plugins/modem-manager/fu-mm-device.c

```diff
@@ -517,13 +517,13 @@ fu_mm_device_io_open_qcdm(FuMmDevice *self, GError **error)
 }
 
 static gboolean
-fu_mm_device_qcdm_cmd(FuMmDevice *self, const guint8 *cmd, gsize cmd_len, GError **error)
+fu_mm_device_qcdm_cmd(FuMmDevice *self, const guint8 *cmd, GError **error)
 {
 	g_autoptr(GBytes) qcdm_req = NULL;
 	g_autoptr(GBytes) qcdm_res = NULL;
 
 	/* command */
-	qcdm_req = g_bytes_new(cmd, cmd_len);
+	qcdm_req = g_bytes_new(cmd, sizeof(cmd));
```

Comment:
are you sure that works? that's doing `sizeof(const guint8 *)`

Reply:
Since `sizeof(cmd[0])` is always 1, I'd say `G_N_ELEMENTS(cmd)`, which was used before, has the same result as `sizeof(cmd)`

Reply:
`8`

Reply:
It's also the size of a 64bit pointer... try adding an extra byte to `emergency_download` to check

Reply:
Ah, I see the problem here... we should't compute the length from the pointer's size